includes/class-wc-stripe-order-handler.php

```diff
@@ -29,6 +29,8 @@ public function __construct() {
 		add_action( 'woocommerce_order_status_cancelled', [ $this, 'cancel_payment' ] );
 		add_action( 'woocommerce_order_status_refunded', [ $this, 'cancel_payment' ] );
 		add_filter( 'woocommerce_tracks_event_properties', [ $this, 'woocommerce_tracks_event_properties' ], 10, 2 );
+
+		add_action( 'wp', [ $this, 'maybe_process_upe_redirect' ] );
```

Comment:
Is there any reasoning for keeping this hook in this Order Handler class as opposed to moving it into the `WC_Stripe_UPE_Payment_Gateway` class instead? I initially thought it was here, because it was going to share the same `wp` action hook as the pre-existing gateway, but it looks like there are two completely separate hooks. If we moved this hook in the UPE Gateway entirely, then we would not need to instantiate the gateway separately. Both these classes extend the old `WC_Stripe_Payment_Gateway`, so I don't think the UPE gateway would lose out on any functionality exclusive to this class.

Reply:
This is the same case as above... the `maybe_process_redirect` for the other gateways is in this `WC_Stripe_Order_Handler`, to keep the code structure consistent I added the new `maybe_process_upe_redirect` here too. I guess it depends on whether we move the intent hooks to the gateway or not, when we do that (either in this PR or in a future one) we also need to move this.

The initial idea was to use the same hook, but it quickly became a big `if` with an even bigger `else` 😢.

Reply:
I can see that sharing the hook could get a little messy. As commented above, let's leave this one on the backburner: it can be resolved in the future along with subscriptions.